src/platform/datapath_raw_xdp_win.c

```diff
@@ -660,6 +670,27 @@ CxPlatDpRawInterfaceInitialize(
             goto Error;
         }
 
+        //
+        // TODO: check whether the interface currently supports the offload
+        // before enabling it (XSK_SOCKOPT_TX_OFFLOAD_CURRENT_CONFIG_CHECKSUM).
```

Comment:
If it's not supported would it simply fail? If so, why check first?

Reply:
Yeah, even I forgot how this works. The app can set `XSK_SOCKOPT_TX_OFFLOAD_CHECKSUM` to enable the checksum offload ring descriptors, which is independent of whether the underlying NIC supports checksum offload and/or has it currently enabled.

The reason for that is the NDIS model (rightly, probably) allows the NIC hardware capabilities and configuration to change on the fly. XDP follows this model by allowing the app to opt in to the checksum ring extensions, even if the NIC doesn't happen to have the offload supported _right now_.

For MsQuic, we'd probably make a simplifying assumption that the NIC capabilities don't change, and enable the ring extension only if the HW currently has the offload enabled, which is the TODO here.